game-core/src/test/java/games/strategy/triplea/delegate/MatchesTest.java

```diff
@@ -254,4 +266,179 @@ void shouldNotMatchWhenSeaTerritoryIsUnownedAndDoesNotHaveTerritoryAttachment()
       assertThat(newMatch(), notMatches(seaTerritory));
     }
   }
+
+  @Nested
+  @ExtendWith(MockitoExtension.class)
+  final class UnitCanBeInBattle {
+
+    @Mock GamePlayer player;
+    GameData gameData;
+
+    @BeforeEach
+    void setupGameData() {
+      gameData = givenGameData().build();
```

Comment:
`givenGameData()` is not as descriptive as one would hope. What is given about this game data? Can we say or require anything to be true about this game data? Though, I suspect the instantiation of any game data is something of a hack for game data being *very* difficult to mock (God class code-smell, we use it everywhere for everything.

I'd recommend renaming this method to `givenAnyGameData()` perhaps to indicate that we require nothing other than an instantiated game data, otherwise any requirements of the data would either be specified as parameters or baked into the game data name. For example, `givenGameDataWithUnits(...)`

Reply:
Looking at this further, perhaps calling it `givenMockGameData()`.

Though, ideally we would not need the object and any data from the game data would individually be passed to this class. In other words, game data is acting as a tuple with other objects. Rather than passing the tuple and grabbing the objects from the tuple, instead we can pass the individual objects from the tuple to where they are needed.

Reply:
I guess `mockGameDataBuilder()` would be a better name for it since that is what it does.  It creates a mock of `GameData` with some basic setup and then allows you to add functionality using a fluent syntax.  It isn't quite a "builder".

I don't think the rename should be part of this PR, though.  It is used in quite a few of my unittests now.  It is how I create mocked GameData objects.